cfgBuilder/cfgBuilder.go

```diff
@@ -0,0 +1,216 @@
+		log.Warn("error closing file", "err", err)
+	}
+	return newFile
+}
+
+func construcEthChainConfig(cfg EthChainConfig, relayer string) RawChainConfig {
+	return RawChainConfig{
+		Name:     cfg.Name,
+		Type:     "ethereum",
+		From:     relayer,
+		Id:       cfg.ChainId,
+		Endpoint: cfg.Endpoint,
+		Opts: EthOpts{
+			BridgeAddress:  cfg.BridgeAddress,
+			Erc20Handler:   cfg.Erc20Handler,
+			Erc721Handler:  cfg.Erc721Handler,
+			GenericHandler: cfg.GenericHandler,
+			GasLimit:       cfg.GasLimit,
+			GasPrice:       cfg.GasPrice,
+			StartBlock:     cfg.StartBlock,
+			Http:           cfg.Http,
+		},
+	}
+}
+
+func constructRelayerConfig(cfg *Config, relayer string) RootConfig {
+	// Create RawConfig structs from the provided Chains
+	var rawCfgs []RawChainConfig
+	for _, chain := range cfg.Chains {
+		raw := construcEthChainConfig(chain, relayer)
+		rawCfgs = append(rawCfgs, raw)
+	}
+
+	return RootConfig{Chains: rawCfgs}
+}
+
+func parseRawConfig(raw *RawConfig) (*Config, error) {
+	var res Config
+	deployer, err := secp256k1.NewKeypairFromString(raw.DeployerPrivateKey[2:])
+	if err != nil {
+		return nil, err
+	}
+	res.Deployer = deployer
+
+	threshold, ok := big.NewInt(0).SetString(raw.RelayerThreshold, 10)
+	if !ok {
+		return nil, fmt.Errorf("failed to parse relayer threshold")
+	}
+	res.RelayerThreshold = threshold
+	res.Relayers = raw.Relayers
+	res.Chains = raw.Chains
+
+	return &res, nil
+}
+
+func parseDeployConfig(path string) (*Config, error) {
+	fp, err := filepath.Abs(path)
+	if err != nil {
+		return nil, err
+	}
+	f, err := os.Open(filepath.Clean(fp))
+	if err != nil {
+		return nil, err
+	}
+
+	var rawCfg RawConfig
+	err = json.NewDecoder(f).Decode(&rawCfg)
+	if err != nil {
+		return nil, err
+	}
+
+	return parseRawConfig(&rawCfg)
+}
+
+// CreateRelayerConfigs takes a prepared config and constructs the configs for each relayer
+func CreateRelayerConfigs(cfg *Config) ([]RootConfig, error) {
+	var configs []RootConfig
+	for _, relayer := range cfg.Relayers {
+		rCfg := constructRelayerConfig(cfg, relayer)
+		configs = append(configs, rCfg)
+	}
+
+	return configs, nil
+}
+
+func main() {
+	// Pares first argument for path
+	if len(os.Args) < 2 {
+		log.Error("Please specify path to config json")
+		os.Exit(1)
+	}
+	path := os.Args[1]
+	if path == "" {
+		log.Error("failed to parse path argument")
+		os.Exit(1)
+	}
+
+	// Read in the config
+	cfg, err := parseDeployConfig(path)
+	if err != nil {
+		log.Error("failed to parse config", "err", err)
+		os.Exit(1)
+	}
+
+	// Construct the individual relayer configs
+	relayerCfgs, err := CreateRelayerConfigs(cfg)
+	if err != nil {
+		log.Error("failed to construct relayer configs", "err", err)
+		os.Exit(1)
+	}
+
+	// Check for output path
+	var outPath string
+	if len(os.Args) == 3 {
+		outPath = os.Args[2]
+	}
+
+	// Write all the configs to files
+	for i, cfg := range relayerCfgs {
+		cfg.ToTOML(filepath.Join(outPath, fmt.Sprintf("config%d.toml", i)))
```

Comment:
should we set outpath default to the chainbridge root directory?

Reply:
It will output to the current directory which I think is expected behaviour. Planning to make this a standalone executable, probably makes a bit more sense in that context.

Reply:
ahh that makes more sense